core/src/main/scala/quasar/destination/gbq/GBQFlow.scala

```diff
@@ -248,15 +254,71 @@ final class GBQFlow[F[_]: Concurrent](
               .as(DestinationError.invalidConfiguration(
                 (GBQDestinationModule.destinationType, 
                   config.sanitizedJson,
-                  ZNEList(s"Error creating GBQ job: ${resp.status.reason}"))).asLeft[Uri])
-
-            
+                  ZNEList(s"Error creating GBQ job: ${resp.status.reason}"))).asLeft[(Uri, GBQJob)])
         }
       }
     })
   }
 
-  private def upload(bytes: Stream[F, Byte], uploadLocation: F[Either[InitializationError[Json], Uri]]): Stream[F, Unit] = {
+  private def getJob(job: GBQJob): Stream[F, GBQJob] = {
+    val jobUrlString =
+      s"https://bigquery.googleapis.com/bigquery/v2/projects/${job.jobReference.projectId}/jobs/${job.id}"
+
+    val jobReq = credsF map { creds =>
+      Request[F](Method.GET, Uri.unsafeFromString(jobUrlString))
+        .withHeaders(creds)
+        .withContentType(`Content-Type`(MediaType.application.json))
+    }
+
+    Stream
+      .eval(jobReq)
+      .flatMap(req => client.stream(req))
+      .evalMap { resp =>
+        resp.status match {
+          case Status.Ok => resp.as[GBQJob]
+          case _         => ApplicativeError[F, Throwable].raiseError[GBQJob](
+                              new RuntimeException(s"getJobStatus failed: ${resp.status.reason}. Check job manually with jobId: ${job.id}" )
+                            )
+        }
+      }
+  }
+
+  private def checkJobStatus(job: GBQJob, timeOut: FiniteDuration, maxRetries: Int, tries: Int): Stream[F, Unit] = {
+
+    def retry: Stream[F, Unit] =
+      Stream.sleep(timeOut) >> getJob(job).flatMap(newJob => checkJobStatus(newJob, timeOut, maxRetries, tries + 1))
```

Comment:
I think what you're trying to do here is "sleep and then getJob", but I'd be very careful with using `>>` to express such composition in case of `Stream`. Because `a >> b` == `a.flatMap(_ => b)`, in case of streams if `a` emits more than one element then your `b` will be executed for every element. In this case it'd work fine because `sleep` is guaranteed to only emit once, but as a rule of thumb I'd recommend expressing `and then` type of logic with `++` when operating on streams.